apps/gk_species_lbo.c

```diff
@@ -30,15 +31,25 @@ gk_species_lbo_init(struct gkyl_gyrokinetic_app *app, struct gk_species *s, stru
   lbo->normNu = false;
   if (s->info.collisions.normNu) {
     lbo->normNu = true;
-    lbo->self_nu_fac = s->info.collisions.self_nu_fac;
+    double nuFrac = s->info.collisions.nuFrac ? s->info.collisions.nuFrac : 1.0;
+    double eps0 = s->info.collisions.eps0 ? s->info.collisions.eps0: GKYL_EPSILON0;
+    double hbar = s->info.collisions.hbar ? s->info.collisions.hbar: GKYL_PLANCKS_CONSTANT_H/2/M_PI;
+    double eV = s->info.collisions.eV ? s->info.collisions.eV: GKYL_ELEMENTARY_CHARGE;
+    struct gkyl_array* bmag_mid_host = app->gk_geom->bmag_mid;
+    if (app->use_gpu) {
+      bmag_mid_host = mkarr(false, 1, 1);
+      gkyl_array_copy(bmag_mid_host, app->gk_geom->bmag_mid);
+    }
+    double *bmag_mid_ptr = gkyl_array_fetch(bmag_mid_host, 0);
+    double bmag_mid = s->info.collisions.bmag_mid ? s->info.collisions.bmag_mid : bmag_mid_ptr[0];
+    if (app->use_gpu)
+      gkyl_array_release(bmag_mid_host);
```

Comment:
@akashukla see earlier comment about more compact way to handle allocate/deallocate bmag_mid_host.